src/hooks/useMemoManager.ts

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useState } from "react";
+import { Memo } from "../types";
+
+// メモの状態を管理するカスタムフック
+export const useMemoManager = () => {
+  const [memos, setMemos] = useState<Memo[]>([]);
+
+  // クライアントサイドでのみローカルストレージにメモを保存
+  useEffect(() => {
+    const storedMemos = localStorage.getItem("memos");
+    if (storedMemos) {
```

Comment:
[SHOULD] 暗黙的ブール変換をやめましょう。

```suggestion
    if (storedMemos !== undefined && storedMemos !== null && storedMemos !== '') {
```